File: javascript/344.reverse-string.js
/*
 * @lc app=leetcode id=344 lang=javascript
 *
 * [344] Reverse String
 */

// @lc code=start
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
  // Solution 1:
  // s.reverse();

  // Solution 2: two pointers
  let left = 0,
    right = s.length - 1;
  while (left < right) {
    [s[left], s[right]] = [s[right], s[left]];
    left++;
    right--;
  }
};
// @lc code=end

File: rust/344.reverse-string.rs
/*
 * @lc app=leetcode id=344 lang=rust
 *
 * [344] Reverse String
 */

// @lc code=start
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        // Solution 1:
        s.reverse();

        // Solution 2: two pointers
        // let (mut left, mut right): (usize, usize) = (0, s.len() - 1);
        // while left < right {
        //     s.swap(left, right);
        //     left += 1;
        //     right -= 1;
        // }
    }
}
// @lc code=end
